raft/src/main/java/org/apache/kafka/raft/ReplicatedLog.java

```diff
@@ -175,6 +180,19 @@ default OptionalLong truncateToEndOffset(OffsetAndEpoch endOffset) {
      */
     Optional<RawSnapshotReader> readSnapshot(OffsetAndEpoch snapshotId) throws IOException;
 
-    default void close() {}
+    // TODO: write documentation
+    Optional<OffsetAndEpoch> latestSnapshotId();
+
+    /**
+     * TODO: document this method
+     * TODO document that this method is thread safe
+     */
+    void snapshotFrozen(OffsetAndEpoch snapshotId);
```

Comment:
This is an implementation detail. We should be able to remove this. The problem is that the `RawSnapshotWriter` implementation lives in the `raft` module while the `ReplicatedLog` implementation lives in the `core` module. To remove this we need to move `FileRawSnapshotWriter` implementation to `core`.

cc @hachikuji